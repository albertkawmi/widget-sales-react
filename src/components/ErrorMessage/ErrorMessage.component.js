import React from 'react'
import Store from '../Store'
import './ErrorMessage.css'

export default class ErrorMessage extends React.Component {
  render() {
    const { error, dismissError } = this.context
    if (!error) return null

    return (
      <div className="error">
        <h4 className="error__heading">Sorry, this page always errors out!</h4>
        <span className="error__text">&nbsp;({error})</span>
        <div className="error__dismiss" onClick={dismissError}>
          <span role="img" aria-label="Dismiss Error">âœ–</span>
        </div>
      </div>
    )
  }
}

ErrorMessage.contextTypes = Store.childContextTypes
