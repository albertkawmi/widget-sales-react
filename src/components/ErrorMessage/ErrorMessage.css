.error {
  display: flex;
  background: tomato;
  border: solid 1px crimson;
  color: white;
  padding: 1em;
}

.error__text {
  flex: 1;
}

.error__dismiss {
  cursor: pointer;
}
