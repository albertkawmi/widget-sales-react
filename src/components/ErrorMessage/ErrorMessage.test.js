/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import ErrorMessage from './ErrorMessage.component'

Enzyme.configure({ adapter: new Adapter() });

describe('<ErrorMessage />', () => {
  it('renders an error message if present', () => {
    const context = { error: 'An error message' }
    const wrapper = Enzyme.shallow(<ErrorMessage />, { context })
    expect(wrapper.text()).toMatch(context.error)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders nothing if there is no error', () => {
    const context = { error: '' }
    const wrapper = Enzyme.shallow(<ErrorMessage />, { context })
    expect(wrapper.text()).toBe('')
  })
})
