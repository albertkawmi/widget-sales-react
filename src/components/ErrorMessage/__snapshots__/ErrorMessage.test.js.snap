// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> renders an error message if present 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorMessage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="error__content"
>
          <h4
                    className="error__heading"
          >
                    Sorry, something went wrong. The status text was: 
                    <span
                              className="error__text"
                    >
                              An error message
                    </span>
          </h4>
</div>,
        <div
          className="error__dismiss"
          onClick={undefined}
>
          <span
                    aria-label="Dismiss Error"
                    role="img"
          >
                    ✖︎
          </span>
</div>,
      ],
      "className": "error wrap",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h4
            className="error__heading"
>
            Sorry, something went wrong. The status text was: 
            <span
                        className="error__text"
            >
                        An error message
            </span>
</h4>,
          "className": "error__content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Sorry, something went wrong. The status text was: ",
              <span
                className="error__text"
>
                An error message
</span>,
            ],
            "className": "error__heading",
          },
          "ref": null,
          "rendered": Array [
            "Sorry, something went wrong. The status text was: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "An error message",
                "className": "error__text",
              },
              "ref": null,
              "rendered": "An error message",
              "type": "span",
            },
          ],
          "type": "h4",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            aria-label="Dismiss Error"
            role="img"
>
            ✖︎
</span>,
          "className": "error__dismiss",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "Dismiss Error",
            "children": "✖︎",
            "role": "img",
          },
          "ref": null,
          "rendered": "✖︎",
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="error__content"
>
            <h4
                        className="error__heading"
            >
                        Sorry, something went wrong. The status text was: 
                        <span
                                    className="error__text"
                        >
                                    An error message
                        </span>
            </h4>
</div>,
          <div
            className="error__dismiss"
            onClick={undefined}
>
            <span
                        aria-label="Dismiss Error"
                        role="img"
            >
                        ✖︎
            </span>
</div>,
        ],
        "className": "error wrap",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4
              className="error__heading"
>
              Sorry, something went wrong. The status text was: 
              <span
                            className="error__text"
              >
                            An error message
              </span>
</h4>,
            "className": "error__content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Sorry, something went wrong. The status text was: ",
                <span
                  className="error__text"
>
                  An error message
</span>,
              ],
              "className": "error__heading",
            },
            "ref": null,
            "rendered": Array [
              "Sorry, something went wrong. The status text was: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "An error message",
                  "className": "error__text",
                },
                "ref": null,
                "rendered": "An error message",
                "type": "span",
              },
            ],
            "type": "h4",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              aria-label="Dismiss Error"
              role="img"
>
              ✖︎
</span>,
            "className": "error__dismiss",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Dismiss Error",
              "children": "✖︎",
              "role": "img",
            },
            "ref": null,
            "rendered": "✖︎",
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "error": "An error message",
    },
  },
}
`;
